src/olympia/git/tasks.py

```diff
@@ -27,18 +31,32 @@ def remove_git_extraction_entry(addon_pk):
 def on_extraction_error(request, exc, traceback, addon_pk):
     log.error('Git extraction failed for add-on "{}".'.format(addon_pk))
 
+    # We only handle *some* errors here because we cannot apply the same
+    # approach to recover from all possible errors. Our current technique to
+    # repair the errors below is to delete the git repository and let the
+    # add-on being re-extracted later.
+    delete_repo = False
+
+    # See: https://github.com/mozilla/addons-server/issues/13590
     if isinstance(exc, BrokenRefError):
-        # We only handle `BrokenRefError` here to recover from such errors and
-        # we cannot apply the same approach for all errors.
-        # See: https://github.com/mozilla/addons-server/issues/13590
+        delete_repo = True
+        log.warning(
+            'Deleting the git repository for add-on "{}" because we detected '
+            'a broken reference.'.format(addon_pk)
+        )
+    # See: https://github.com/mozilla/addons-server/issues/14127
+    if isinstance(exc, MissingMasterBranchError):
```

Comment:
Both of these cases do the exact same thing, with the only difference being the message logged. Would it make more sense to combine them into a single case which executes if either exception occurs, and if it's really important to have different log messages then deal with that when building the log message?

Reply:
I could not find a way to have different log messages in a cleaner way and those logs are very important for debugging.